core/events/events_procedure_change_return.ts

```diff
@@ -5,19 +5,39 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import {IProcedureModel} from '../interfaces/i_procedure_model.js';
+import type {IProcedureModel} from '../interfaces/i_procedure_model.js';
 import * as registry from '../registry.js';
-import {Workspace} from '../workspace.js';
+import type {Workspace} from '../workspace.js';
 
 import {ProcedureBase} from './events_procedure_base.js';
 import * as eventUtils from './utils.js';
 
 
 export class ProcedureChangeReturn extends ProcedureBase {
+  type = eventUtils.PROCEDURE_CHANGE_RETURN;
+  private newTypes: string[]|null;
+
   constructor(
       workpace: Workspace, model: IProcedureModel,
```

Comment:
I appreciate that changing the existing API is probably not sensible, but:

How does it work if you want to run an event from one workspace in another?  Wouldn't it make more sense for `workspace` to be a parameter to `run`?

Reply:
To run an event in another workspace you have to serialize the event and deserialize it into the new workspace. The event system was originally meant for transmitting events over the wire, not running them in different workspaces on the same page. So no one thought about the usecase of passing a workspace to `run`. And at this point, yeah probably not worth changing the API.